src/index.ts

```diff
@@ -2667,6 +2661,13 @@ async function applyInternalTx(
     //   network.timestamp = txTimestamp
     //   devAccount.timestamp = txTimestamp
     // }
+
+    // eslint-disable-next-line security/detect-object-injection
+    const addressHash = wrappedStates[networkAccount].stateId
+
+    const ourAppDefinedData = applyResponse.appDefinedData as OurAppDefinedData
+    // network will consens that this is the correct value
+    ourAppDefinedData.globalMsg = { address: networkAccount, addressHash, value, when, source: value.from }
```

Comment:
`addressHash` sounds misleading. It would be better to call it `stateHash` but this is only a suggestion and do consider the effort before making this change.